doc/api/signed-requests.md

```diff
@@ -0,0 +1,127 @@
+
+The signature process is broken down into small steps for this explanation.  First, you create a date stamp.  Next, some signed headers are formatted.  Next, some content that will be signed.  The [HMAC] of the content is calculated.  Lastly, the request has the right headers set.
+
+
+Date Stamp
+----------
+
+Create a date stamp in [ISO 8601] format.  Specifically, we're looking for `yyyy-mm-ddThh:mm:ssZ` format.  It is used for an `X-OpenToken-Date` header.  Our example will use this date.
+
+    2016-11-17T19:53:23Z
+
+This date must be within a few minutes of the current time, so generate this date right before you submit the request.
+
+
+Signed Headers
+--------------
+
+When sending a signed request, the server will only honor the headers that are signed.  This is to help prevent proxies from inserting additional headers or manipulating the request.  The `Host` header helps ensure the client knows where it is sending requests instead of having the requests redirected somehow.  The `Content-Type` header confirms that content isn't being injected or manipulated and is especially important for [tokenization](tokens.md).  Finally, an `X-OpenToken-Date` header is used to prevent replay attacks.
+
+Let's format these headers now.  Order here must match the order specified in the `Authorization` header, discussed later.  All header names must be changed to lowercase.  Trim the header name by removing all whitespace (spaces, tabs, etc.) from the beginning and end.  Next, append a colon.  Finally, trim the header value but do not change it to all lowercase.
+
+One special exception is that the `Host` header's value must be lowercase.  DNS is not case-sensitive and the `Host` header in the request could get mangled by a proxy.
+
+The `X-OpenToken-Date` header's value will be the date stamp generated in the earlier step.
+
+Join all the headers together with newlines between them.  They should look like our example.
+
+    host:api.opentoken.io
+    content-type:text/plain
+    x-opentoken-date:2016-11-17T19:53:23Z
+
+Other headers may be signed as well.  The three listed are mandatory.
+
+
+Generate the Signing Content
+----------------------------
+
+In order for the signatures to match, both the server and the client must be signing the exact same content.  This content follows the following template.  Please note that all newline characters are Unix newlines, hex 0x0a.
+
+    HTTP Verb + Newline
+    URI Path + Newline
+    Query String + Newline
+    Formatted Headers + Newline
+    Newline
+    Body
+
+The HTTP verb must be in all uppercase.
+
+The URI path is everything after the hostname, including the first slash.  This must preserve the case.
+
+The query string includes any optional parameters.  In URIs they appear after a `?`.  If you had a URI such as `https://api.opentoken.io/test?something=true`, then this line would be `something=true`.  It would not contain the `?`.
```

Comment:
What if there is more than one query parameter? I know it's pretty simple as it would be something=true&somethingElse=true, but I think we should call it out just in case.